core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.controller.ts

```diff
@@ -100,6 +142,68 @@ angular.module('oppia').controller('TranslationModalController', [
       $scope.activeWrittenTranslation.html = '';
     };
 
+    $scope.getElementTexts = function(elements, type) {
+      const textWrapperLength = 6;
+      const attributes = Array.from(elements, function(element: HTMLElement) {
+        if (element.localName === 'oppia-noninteractive-image') {
+          const attribute = element.attributes[type].value;
+          return attribute.substring(
+            textWrapperLength, attribute.length - textWrapperLength);
+        }
+      });
+      return attributes.filter(attribute => attribute);
+    };
+
+    $scope.copiedAllElements = function(originalElements, translatedElements) {
+      const hasMatchingTranslatedElement = (element) => (
+        translatedElements.includes(element));
+      return !originalElements.every(hasMatchingTranslatedElement);
+    };
+
+    $scope.changedImgDetails = function(originalElements, translatedElements) {
+      if (originalElements.length === 0) {
+        return false;
+      }
+      const hasMatchingTranslatedElement = (element) => (
+        translatedElements.includes(element) && originalElements.length > 0);
+      return originalElements.every(hasMatchingTranslatedElement);
```

Comment:
Sorry, actually this should be `Arrays.some()`, right? Since we want to raise an error if *any* of the alt texts or descriptions are the same (rather than if all them are the same)? Is there a way to add a test case for this?

Reply:
Done